modules/admindebug.cpp

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2004-2018 ZNC, see the NOTICE file for details.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <znc/FileUtils.h>
+#include <znc/Server.h>
+#include <znc/IRCNetwork.h>
+#include <znc/User.h>
+#include <znc/ZNCDebug.h>
+
+class CAdminDebugMod : public CModule {
+  public:
+    MODCONSTRUCTOR(CAdminDebugMod) {
+        AddHelpCommand();
+        AddCommand("Enable", "", t_d("Enable Debug Mode"),
+                   [=](const CString& sLine) { CommandEnable(sLine); });
+        AddCommand("Disable", "", t_d("Disable Debug Mode"),
+                   [=](const CString& sLine) { CommandDisable(sLine); });
+        AddCommand("Status", "", t_d("Show the Debug Mode status"),
+                   [=](const CString& sLine) { CommandStatus(sLine); });
+    }
+
+    bool OnLoad(const CString& sArgs, CString& sMessage) override {
+        if (GetUser() != nullptr) {
+            return GetUser()->IsAdmin();

```

Comment:
How about an error message explaining why loading was denied? Also, have you tested this? I think global modules are loaded with `GetUser() == nullptr`.

Reply:
>Anyone can load the module, presumably. (see #1556 (review))
The idea was to deny loading of the module unless it's on a global level by an administrator, GetUser() always seemed to be null. It doesn't accomplish anything, I was just trying to satisfy the "OnLoad returns a bool" condition, because it seems pointless to always return true for a module mainly designed for admins.

Reply:
You don't do it to "satisfy some weird requirement"; that requirement is there for a reason - to block loading the module if needed. You can simply drop the whole `OnLoad`, it already returns `true` by default.
This module is global; it can't be loaded by anyone other than admins anyway.

> Anyone can load the module, presumably. (see #1556 (review))

What exactly are you linking at?

Reply:
@psychon 
Remnants from https://github.com/znc/znc/blob/master/modules/adminlog.cpp#L42-L58 (because that's the module I copied from.) I will change.

@DarthGandalf I was linking to https://github.com/znc/znc/pull/1556#discussion_r195183328